### PR TITLE
DOPS-425/ldc-still-starts-up-when-stunnel-fails-to-start-on-heroku

### PR DESCRIPTION
**Pull request by alexebird**
@bmangold @lumengxi @alexebird @mike809 @netwerker

### CODE REVIEW

File: test/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM ubuntu
+USER root
+RUN apt-get update -y
```
Comment (lumengxi) [inline test/Dockerfile]: Nit: Lets consolidate these `RUN`s to have a smaller docker image.

**Comment by lumengxi** [comment]
cc @jakeyr this PR is to make sure buildpack exit early if stunnel failed to start

File: README.md [hunk 2]
```diff
@@ -1,2 +1,20 @@
...
+aux-start stunnel SIGINT /bin/bash -c 'echo fake-stunnel-start ; sleep 2 ; echo fake-stunnel-crash ; exit 1'
+# stunnel "runs" fine
+aux-start stunnel SIGINT /bin/bash -c 'echo fake-stunnel-start ; sleep 1000 ; echo fake-stunnel-crash'
+```
+
+```
+docker build -t lendup-stunnel .
+touch bash_history
+docker run --name=stunnel --rm -it -v ${PWD}/bash_history:/root/.bash_history lendup-stunnel
+container$ app/bin/start-stunnel /bin/bash -c 'echo starting-app.......................... ; sleep 100000000'
```
Comment (bmangold) [inline README.md:19]: would be awesome to add in BATS sometime for this (fairly easy) : https://github.com/sstephenson/bats

**Comment by bmangold** [comment]
LGTM

**Comment by alexebird** [comment]
Can we test this before it goes to prod?

**Pull request closed without merging**